finagle-base-http/src/main/scala/com/twitter/finagle/http/CookieMap.scala

```diff
@@ -157,17 +162,21 @@ class CookieMap private[finagle] (message: Message, cookieCodec: CookieCodec)
   private[this] def addNoRewrite(name: String, cookie: Cookie): Unit = {
     val prev = underlying(name)
     val next =
-      if (prev.contains(cookie)) cookie :: prev.filter(c => c != cookie)
+      if (message.isResponse && prev.contains(cookie)) cookie :: prev.filter(c => c != cookie)
```

Comment:
question: we still want to prevent illegal duplicative cookies (eg ones with the same path, domain, and name), right?  I'm a bit worried that we might be overreaching here.  can we ensure that we continue to filter those out?

Reply:
@guilgaly bumping this once more :)